### ISSUE TITLE
Unresolved Reference in Bashed Patch

### ISSUE THREAD

**Issue by Sharlikran** [wrye-bash/wrye-bash#346]
If I use Wrye Bash with classic Skyrim this plugin [No Vampire Attacks In Towns.zip](https://github.com/wrye-bash/wrye-bash/files/587532/No.Vampire.Attacks.In.Towns.zip) will be assigned the proper FormID ([as shown here](http://i.imgur.com/u9Z7JBM.jpg)) in relation to it's master, which is Dawnguard.esm.  

The code as of commit https://github.com/wrye-bash/wrye-bash/commit/aedf071bb25d20721affec2e80f1add6c51f6e76 when used with Skyrim SE makes two of the FormIDs in those records end up with a prefix of 02 ([as shown here](http://i.imgur.com/F0JqZuC.jpg)) when it should be in line with the Dawnguard master and have a prefix of 01.  Which causes an unresolved reference in the Bashed Patch.

When the FormID prefix is assigned properly then it matches the Master in the file.  When that happens Wrye bash or Skyrim SE can properly renumber the FormIDs based on the plugin's position in the load order.  When it can't do that then bad things will happen ([as shown here](http://i.imgur.com/ewxn5FR.jpg)) as the FormID table isn't being properly managed.

I can't provide Git bisect info because this only happens with the Star Load order system.

I am wondering.  Do you have an array for loadorder.txt and separate array for plugins.txt for classic Skyrim?  Do you use the same two separate arrays for Skyrim SE and the Star Load order?

**Comment by Utumno** [comment]
Please test with latest skyrim se support branch

**Comment by Sharlikran** [comment]
Yes still happens with [skyrim-se-support](https://github.com/wrye-bash/wrye-bash/tree/skyrim-se-support) at commit e3a0c7f2561c160e9fa7a05cc8ff0aad62e0e259.

**Comment by Utumno** [comment]
Please give me as detailed steps to reproduce as possible - my tesxedit skills are rusty :D

**Comment by Sharlikran** [comment]
Just to reiterate this is not for Skyrim SE.  Do you have all the Skyrim DLC for the original version?  You will need at least the Skyrim Dawnguard DLC.

1. Deactivate all patchers except Merge Patches

![mergepatchesreproduce](https://cloud.githubusercontent.com/assets/1912426/20250263/8d7fad74-a9c1-11e6-8d65-d0a0c4f919e4.jpg)
2. Use the file provided.  It is the most minimal reproducer I feel I can provide.
3. Build a Bash Patch.

I can only offer what I think you would need to look for and I can't condense it or provide it in an extremely technical way.  This is the best I can do.  While watching what happens during the bash patch build, for classic Skyrim I would suspect it will start to build a Bash Patch.

As it does so and begins to merge those records from No Vampire Attacks In Towns.esp into the Bash Patch.  I suspect Skyrim.esm will be added.  Even though no records reference it yet as that's the main game file.  Then once it determines it will need to add the master Dawnguard.esm, it will be assigned to the Bashed Patch, 0.esp header.  The Master index will be as follows.

00 Skyrim
01 Dawnguard.esm
02 Bashed Patch, 0.esp

Obviously the Bashed Patch, 0.esp is not part of the header but the 02 indicates all the records native the the Bashed Patch only.

You would have a layout like this if you have [15 plugins](http://i.imgur.com/Ycc8xZx.jpg).

So when it builds the Bashed Patch for Skyrim I can use a hex editor and I see that Wrye Bash added the FormID with a master index of 01.  When I open the Bashed Patch in TES5Edit it properly aligns with the record in Dawnguard.esm.  If I load a Bashed Patch into SSEEdit for Skyrim SE it is assigned the wrong master index.  In the Hex editor it received the next subsequent number available which was 02.

When more plugins are involved for the SSE Bashed Patch the number is...  Well I was going to tell you but the code is broken.

**Comment by Sharlikran** [comment]
I hid Wrye Bash SSE on the Nexus.  I don't know how to proceed.  I will add me code to my fork but it will not be rebased in any way.  Sorry about that.

**Comment by Utumno** [comment]
> Just to reiterate this is not for Skyrim SE

You mean the procedure you describe but the bug is in SE right ?

> I will add me code to my fork but it will not be rebased in any way. Sorry about that.

Do and I will rebase it myself when I get some time.

No panic please it's just a bug, strange of course.

**Comment by Sharlikran** [comment]
CTDA may be broken may not.  There are no union deciders but the reason you shouldn't look into that is because this works for classic Skyrim with the exact same file.

> You mean the procedure you describe but the bug is in SE right ?

Correct.  The procedure I am trying to get you to do so simple the explanation is probably what's holding you up.  Not sure.  

I'm trying to get you to just download the  No Vampire Attacks In Towns.zip.  Did you get that far yet? (I mean that in a polite literal sense)  Then parse through it and when you come across places where it is adding the record to the Bash patch you will see that the form ID will end as follows and it should be exact.

For EDID: DLC1VQ00VampireAttackNodePM and the entry DLC1VQMinLevelVampireAttacks the FormID will be exactly 01019942 which is a GLOB record.  For DLC1EclipseAttackNextChanceNight 010145E9 which is also a GLOB record.

For EDID: DLC1VQ00VampireAttackNodeAM and the entry DLC1VQMinLevelVampireAttacks the FormID will be exactly 01019942 which is a GLOB record.  For DLC1EclipseAttackNextChanceNight 010145E9 which is also a GLOB record.

The two FormIDs are the same for both.  They should be assigned [01] as the FormID prefix, should point to Dawnguard.esm in the Bashed Patch header, not 02 which is usually the Actual position of Dawnguard.esm in the loaded plugins.

When I do this for Skyrim SE the FormID gets assigned a 02 from the entire load order or loadorder.txt or plugins.txt and not 01 which is the Master Index of the Bashed Patch.  You probably assign it correctly for Oblivion and classic Skyrim but for the Start Load order have some ```if thisMode then``` and you skip something or think it has to be done differently.

---------------------------
### long winded stuff

What I am trying to get you to look for is how you assign all of that for the Star Load order.  I am wondering if this happens because you have a check like ```if fsName == Fallout4:``` or ```if starLoadOrder:``` because for Fallout 4 you might be trying to handle the load order array with fewer variables thinking it's mroe efficent.  Use the same variables you use for Oblivion and/or clasic Skyrim.  

The loadorder.txt is the entire data folder and plugins.txt is the active mods.  Well for Skyrim SE and Fallout 4 do you use the same two variables?  All the time?  The entire load order is all of plugins.txt just strip out the star "*".  The active mods are only the ones with the star as the first char just strip out the "*" again.

Because I'm thinking the disconnection is that for Skyrim you properly assign the right FormID to the header of the Bashed Patch and the records because you see it in your mind and then made the code work for the load order much more clearly.  You were able to use the Long Fid conversion that already existed for Oblivion.  Where for Wrye bash, although there are Short Fids also, what makes Wrye Bash work is it doesn't see FormIDs as 010145E9 it sees them as ("Dawnguard.esm", 0145E9) so you always get the FormID correct.  

However remeber that ("Dawnguard.esm", 0145E9) to all the loaded plugins might be 02, but to the Bashed Patch it is going to be the Master Index of just the Bashed Patch or possibly 01.

Where the error occurs you are simply using the wrong array.  Or at least that's what I am thinking.  By using the wrong array or by skipping something you feel you can skip since Fallout 4, you are not using the same array you used for clasic Skyrim and loadorder.txt.  Then when you go through the code and ("Dawnguard.esm", 0145E9) comes up for Skyrim SE, the FormID gets a number higher then it should.  You don't want to do anything stupid like (xNumber -1) or anything like that.  

---------------
### somewhat short summary

When using your Star Load order variables, somehow the FormID assigned is from what you are calling the entire load order or all the files loaded at that time which makes the FormID end up starting with 02 since Dawnguard.esm is at position 02 relative to all the loaded plugins.

What it should be assigned is Dawnguard.esm relative to the Bashed Patch which since at the time the Bashed Patch adds the master is after 00 Skyrim.esm so Dawnguard becomes 01 Dawnguard.esm.

If you add other masters, all the other masters get he next number like 02 Hearthfires.esm.  When you finalize the Bashed Patch all the FormIDs have to convert from ("Dawnguard.esm", 0145E9) to [xx]0145E9 where [xx] is the Master Index of the Bashed Patch, not all the loaded plugins.  Like how I try to show, but do a crappy job of it, in [this screen shot](http://i.imgur.com/Ycc8xZx.jpg).

**Comment by Utumno** [comment]
Will have a look certainly - just be aware that the bashed patch code has nothing to do with the load order code as it should (it's called modularity). So the same load order that is displayed in Bash is the load order the patcher gets irrespective of game

**Comment by Utumno** [comment]
Well - first of all the bashed patch blows internally as it can't extract the string files - so we disable bashed patch till the string files load. DIdn't you see those messages ?

```
__init__.py 2736 getStringsPaths:    Error loading BSA file: Skyrim - Sounds.bsa
Traceback (most recent call last):
  File "bash\bosh\__init__.py", line 2733, in getStringsPaths
    bsaFile = libbsa.BSAHandle(path)
  File "bash\libbsa.py", line 301, in __init__
    _Cbsa_open(byref(self._handle),_enc(path))
  File "bash\libbsa.py", line 215, in LibbsaErrorCheck
    raise LibbsaError(result)
LibbsaError: LibbsaError: LIBBSA_ERROR_PARSE_FAIL:Structure of "F:\Games\The Elder Scrolls V Skyrim Special Edition\Data\Skyrim - Sounds.bsa" is invalid.
```

And so on

Didn't your Bash patch log have this:

```
Bashed Patch, 0.esp
•  Overview
 
Overview
 
Date/Time
•  11/14/2016 4:01:36 PM
•  Elapsed Time: 0:00:00.574
 
Load Error Mods
The following mods had load errors and were skipped while building the patch. Most likely this problem is due to a badly formatted mod. For more info, see Bashed Patch: Error Messages. 
•  Skyrim.esm: Skyrim.esm: Could not locate Strings File 'Skyrim_ENGLISH.DLSTRINGS'
•  Update.esm: Update.esm: Could not locate Strings File 'Update_ENGLISH.DLSTRINGS'
•  Dragonborn.esm: Dragonborn.esm: Could not locate Strings File 'Dragonborn_ENGLISH.ILSTRINGS'
•  Dawnguard.esm: Dawnguard.esm: Could not locate Strings File 'Dawnguard_ENGLISH.ILSTRINGS'
•  Hearthfires.esm: HearthFires.esm: Could not locate Strings File 'HearthFires_ENGLISH.ILSTRINGS'
 
Active Mods
•  00 Skyrim.esm
•  01 Update.esm
•  02 Dragonborn.esm
•  03 Dawnguard.esm
•  04 Hearthfires.esm
•  ++ No Vampire Attacks In Towns.esp
 
```


Now if it blows like that all sort of stuff can go wrong - so we have to extract the skyrim se bsas and then enable the patcher.

Which is a lot of tricky technical work so don't hold your breath

**Comment by Sharlikran** [comment]
## Branch used

HEAD : [skyrim-se-support](https://github.com/wrye-bash/wrye-bash/tree/skyrim-se-support) Branch: [Sharlikran-Wrye : MyBranch](https://github.com/Sharlikran/wrye-bash/tree/shar-fix-borked-code)

I used your code form skyrim-se-support but I copied in the exact skyrim folder then changed the required constants for Skyrim SE.  So both tests are essentially your code.  I obviously extract the strings files for Skyrim SE of Fallout 4.

## Skyrim SE Resutls

----------------

Bashed Patch, 0.esp
•  Overview

Overview

Date/Time
•  11/14/2016 7:41:08 AM
•  Elapsed Time: 0:00:06.033

Active Mods
•  00 Skyrim.esm
•  01 Update.esm
•  02 Dawnguard.esm
•  03 Hearthfires.esm
•  04 Dragonborn.esm
•  05 Campfire.esm
•  06 3rdEraWeaponsMoS.esm
•  07 Unofficial Skyrim Special Edition Patch.esp [Version 4.0.2]
•  08 3rdEraWeapMoS-lvlList.esp
•  09 GodGear.esp
•  0A LrsamwaysExpandedSkyrimWeaponry.esp
•  0B When Vampires Attack.esp [Version 3.0]
•  0C RealisticWaterTwo.esp
•  0D Holidays.esp
•  0E Dovahkiin_Keep.esp [Version 3.0]
•  0F Run For Your Lives.esp [Version 3.0]
•  10 Dovahkiin_Retreat.esp [Version 1.0]
•  11 TrueStormsSE.esp
•  12 DiverseDragonsCollectionSE.esp
•  13 Hideout_DLC_Main.esp [Version 10.4]
•  14 Riftrun_HF_v_2_0.esp [Version 2.0]
•  15 SofiaFollower.esp
•  16 WetandCold.esp
•  ++ No Vampire Attacks In Towns.esp

Log File:
```Python
Wrye Bash starting
Using Wrye Bash Version 307
OS info: Windows-7-6.1.7601-SP1
Python version: 2.7.8
wxPython version: 2.8.12.1 (msw-unicode)
input encoding: None; output encoding: None; locale: ('en_US', 'cp1252')
filesystem encoding: mbcs 
Searching for game to manage:
bush.py   71 _supportedGames: Detected the following supported games via Windows Registry:
bush.py   73 _supportedGames:  oblivion: G:\Games\steamapps\common\Oblivion
bush.py   73 _supportedGames:  skyrim: F:\Games\steamapps\common\Skyrim
bush.py   73 _supportedGames:  skyrim special edition: F:\Games\steamapps\common\Skyrim Special Edition
bush.py   73 _supportedGames:  fallout4: C:\Games\steamapps\common\Fallout 4
bush.py  131 _detectGames: Detecting games via the -o argument, bash.ini and relative path:
bush.py  137 _detectGames: Set game mode to skyrim special edition found in parent directory of Mopy:  F:\Games\steamapps\common\Skyrim Special Edition
bush.py  172 setGame: No preferred game specified.
bush.py  152 __setGame:  Using skyrim special edition game: F:\Games\steamapps\common\Skyrim Special Edition
testing UAC
__init__.py 7817 initDirs: Using libbsa API version: 2.0.0
mods_metadata.py  220 __init__: Using LOOT API version: 0.6.0
mods_metadata.py  226 __init__: The LOOT API does not support the current game.
Exception AttributeError: "'LootDb' object has no attribute '_DB'" in <bound method LootDb.__del__ of <bash.loot.LootDb object at 0x070441D0>> ignored
```
I expect the LOOT error it can be ignored.  If I add the proper LOOT support that goes away.

## classic Skyrim Resutls

----------------

Bashed Patch, 0.esp
•  Overview

Overview

Date/Time
•  11/14/2016 7:45:54 AM
•  Elapsed Time: 0:00:08.709

Active Mods
•  00 Skyrim.esm
•  01 Update.esm
•  02 Dawnguard.esm
•  03 HearthFires.esm
•  04 Dragonborn.esm
•  05 Unofficial Skyrim Legendary Edition Patch.esp [Version 3.0.6]
•  06 LeatherArmorConsealment.esp
•  07 Executor.esp
•  08 Dovahkiin_Retreat.esp [Version 1.0]
•  09 Dovahkiin_Keep.esp [Version 2.0]
•  0A Riftrun_HF_v_2_0.esp [Version 2.0]
•  0B Hideout_DLC_Main.esp [Version 10.4]
•  0C SofiaFollower.esp
•  ++ No Vampire Attacks In Towns.esp

Log File:
```Python
Wrye Bash starting
Using Wrye Bash Version 307
OS info: Windows-7-6.1.7601-SP1
Python version: 2.7.8
wxPython version: 2.8.12.1 (msw-unicode)
input encoding: None; output encoding: None; locale: ('en_US', 'cp1252')
filesystem encoding: mbcs 
Searching for game to manage:
bush.py   71 _supportedGames: Detected the following supported games via Windows Registry:
bush.py   73 _supportedGames:  oblivion: G:\Games\steamapps\common\Oblivion
bush.py   73 _supportedGames:  skyrim: F:\Games\steamapps\common\Skyrim
bush.py   73 _supportedGames:  skyrim special edition: F:\Games\steamapps\common\Skyrim Special Edition
bush.py   73 _supportedGames:  fallout4: C:\Games\steamapps\common\Fallout 4
bush.py  131 _detectGames: Detecting games via the -o argument, bash.ini and relative path:
bush.py  137 _detectGames: Set game mode to skyrim found in parent directory of Mopy:  F:\Games\steamapps\common\Skyrim
bush.py  172 setGame: No preferred game specified.
bush.py  152 __setGame:  Using skyrim game: F:\Games\steamapps\common\Skyrim
testing UAC
__init__.py 7817 initDirs: Using libbsa API version: 2.0.0
mods_metadata.py  220 __init__: Using LOOT API version: 0.6.0
mods_metadata.py  268 refreshBashTags: An error occurred while parsing file C:\Users\DanoPDX\AppData\Local\LOOT\Skyrim\masterlist.yaml:
Traceback (most recent call last):
  File "bash\bosh\mods_metadata.py", line 264, in refreshBashTags
    lootDb.Load(path.s)
  File "bash\loot.py", line 342, in Load
    self.PlainLoad(masterlist, userlist)
  File "bash\loot.py", line 346, in PlainLoad
    _CLoad(self._DB, _enc(masterlist), _enc(userlist) if userlist else None)
  File "bash\loot.py", line 234, in LootErrorCheck
    raise LootError(result)
LootError: LootError: LOOT_ERROR_PARSE_FAIL:yaml-cpp: error at line 322, column 9: bad conversion
```

**Comment by Sharlikran** [comment]
> Will have a look certainly - just be aware that the bashed patch code has nothing to do with the load order code as it should (it's called modularity). So the same load order that is displayed in Bash is the load order the patcher gets irrespective of game

Then this is something else and I wouldn't know where to start.  The problem is if the bashed patch code has nothing to do with the load order code then since almost every single record is the same between Skyrim and Skyrim SE then the Bashed Patch should be identical.  They aren't.

[Clasic Skyrim Bashed Patch, 0.zip](https://github.com/wrye-bash/wrye-bash/files/589751/Clasic.Skyrim.Bashed.Patch.0.zip)

[Skyrim SE Bashed Patch, 0.zip](https://github.com/wrye-bash/wrye-bash/files/589753/Skyrim.SE.Bashed.Patch.0.zip)

There are some places that have a lot of noise in them but in a side by side comparison you get this.

![comparedbashpatch](https://cloud.githubusercontent.com/assets/1912426/20271747/9fdf3956-aa40-11e6-928f-1a30f144641b.jpg)

So why would the exact same code, with the exact same record definition, with the exact same functions, number the FormID 01019942 for classic Skyrim but  02019942 for Skyrim SE.

That doesn't seem like it's the missing strings files to me.

**Comment by Utumno** [comment]
I get the bashdump and BP log I posted with only  • ++ No Vampire Attacks In Towns.esp in my load order apart from vanilla on skyrim-se-support head 33e6ee7022ef85313e0665da67836ccbe594af84

Did you manually tag anything ?

**Comment by Sharlikran** [comment]
No

**Comment by Sharlikran** [comment]
What do you get with classic skyrim?

**Comment by Sharlikran** [comment]
@lojack5 My suspicions were correct.  The situation is caused by the many places in the Wrye Bash code where an integer, Float, or FormID can exist depending on a certain condition.  2 or 3 years ago when we spoke on Skype you mentioned that in many places things are done as a catch all.  In this case it wasn't a good idea.  

I was able to resolve the issue by checking the same bit that xEdit does in [wbCTDACompValueDecider](https://github.com/TES5Edit/TES5Edit/blob/sharlikran-fo4dump/wbDefinitionsTES5.pas#L3124) and looking for the UseGlobal flag defined in [wbCtdaTypeToStr](https://github.com/TES5Edit/TES5Edit/blob/sharlikran-fo4dump/wbDefinitionsTES5.pas#L1439-L1452) and after doing so because the FormID is added to the Fid table now the FormID is renumbered correctly.

Lojack, Can you tell me, how will this affect a Float when it is read as an Integer?  If it's just read in as Hex 9A 99 99 3E which is 0.300000, won't it still be written as Hex 9A 99 99 3E and then xEdit could still treat it as a Float?

If not then how would Wrye Bash change ```def __init__(self):``` in class MelCTDAHandler as I discuss in this commit https://github.com/Sharlikran/wrye-bash/commit/614a5a4da63f965210e13a34c13d27552cb5c28e or you may view [class MelCTDAHandler](https://github.com/Sharlikran/wrye-bash/blob/614a5a4da63f965210e13a34c13d27552cb5c28e/Mopy/bash/game/skyrim/records.py#L171-L337) in its entirety.

**Comment by Arthmoor** [comment]
Seems a contentious issue, but... I tested this in SSE which is where the problem seems to be showing up. I am unable to reproduce this at all. The form IDs for the GLOB values are coming out as expected and do not display errors when viewed in xEdit.

**Comment by Sharlikran** [comment]
So you used the No Vampire Attacks In Towns.esp I included in the Zip file I listed?  You set it to merge into the bashed patch, and disabled all other patchers?  That should give you only two records.  When you load only the Bashed Patch and No Vampire Attacks In Towns.esp in SSEEdit you didn't see the unresolved references I [show here](http://i.imgur.com/F0JqZuC.jpg).

**Comment by Arthmoor** [comment]
Yep. Used the file you linked, set Bash to only merge the one file in, disabled all other patcher for that run. The resulting patch is 2 SMQN records. Neither of them has any unresolved referenced to GLOB records.

**Comment by Sharlikran** [comment]
Well that's odd because previously no matter what I did the FormID was incorrect.  The situation was worse with my entire load order as shown in my [old screen shot](http://i.imgur.com/ewxn5FR.jpg).

I just tested it all again because I know I'm not making this up and now with utumno-wip the FormID is incorrect.  With my branch [wrye-MelCTDAHandler](https://github.com/Sharlikran/wrye-bash/tree/wrye-MelCTDAHandler) based off of utumno-wip the FormID is correct.  

What has me pulling out my hair out now is that my old code which is in my branch [dev-all-games](https://github.com/Sharlikran/wrye-bash/tree/dev-all-games) even with my fix, the FormID is incorrect.  I don't understand this now.  I'm going to bed.  I spent most of my day trying to figure this out.

**Comment by Arthmoor** [comment]
Dunno, I tried it with a full load order too and it still reflects the proper value for those GLOBs. No strange morphing or anything.

**Comment by Sharlikran** [comment]
Arthmoor, I still use older Python because of Valda's FO3 and FNV Wrye Flash versions.  Does your Bash Log start with this?

```Python
Wrye Bash starting
Using Wrye Bash Version 307.37
OS info: Windows-7-6.1.7601-SP1
Python version: 2.7.8
wxPython version: 2.8.12.1 (msw-unicode)
input encoding: None; output encoding: None; locale: ('en_US', 'cp1252')
filesystem encoding: mbcs 
Searching for game to manage:
```
Do you have windows 10 or windows 7?  I have windows 7.

**Comment by Utumno** [comment]
Python 2.7.8 ? I am sorry that's not supported

Sharlikran you went ahead and completely ignored the dev branch - you even committed stuff that is not on the dev branch like leandor's cbash 7. So you are asking support for a different program.

Bug reports in here refer to the current dev code at most, not to experimental branches and in particular not to experimental branches I have not written myself.

This is the last time I will say it ever: you can help - delete your branches and reapply your changes over the upstream dev branch so I can compare them with my changes easily.

Till then closing this as invalid

**Comment by Sharlikran** [comment]
> not to experimental branches and in particular not to experimental branches I have not written myself.

So I can't test utumno-wip and then report bugs with it?  I'm confused?  I used your utumno-wip branch and my log even said 307.201611190020 did you notice that?

**Comment by Arthmoor** [comment]
I'm using Python 2.7.12 on Windows 7. I think if the issue is that you're using an old Python release, you need to update that instead of trying to fight it.

**Comment by Sharlikran** [comment]
> Bug reports in here refer to the current dev code at most, not to experimental branches

I'll just wait until this is merged into dev and test it again if that's the standard procedure.  

Odd though.  I just posted that I had updated to Python 2.7.10 according to the Wrye Bash Advanced Readme.html.  Then using the utumno-wip branch that I still have the incorrect FormID.  I didn't know that 2.7.12 was required.  I'm going to print this page in Windows .xps format and forward it to @lojack5 because he is the one that told me it's important to make sure all FormIDs are accounted for.  Which is also true for xEdit.

As I stated when I spoke to Lojack many years ago I mentioned places on the code that don't decide when it's a FormID and when it's an integer.  We were talking about MelCoed specifically and he mentioned.  He wrote the MelCoed class and told me then that it would be fine the way it was and that it could be looked at later.  Again that was when he was still around to help out and offer advice.

**Comment by Utumno** [comment]
Well I there is probably a bug here - and it's probably not related to the python version (although other things will break in 2.7.8)

And testing my wip branch is of course welcome I never said otherwise - read above

What I kept repeating for the past years is that I can't follow code that diverges from the dev or my work - which is self evident and repeated millions of times. And despite what you said you went ahead and made all your edits _in the copy pasted code from the skyrim constants and records_.

You even created a new dev with commits that are not in this dev

You should have taken my  skyrim se branch and applied your edits over that - then it would be a piece of cake for me to see what's wrong. I spend all my friday comparing your branch with mine, deleting duplicate code with skyrim and trying to see what changed - most of your stuff _is already in dev_ like the edits in skyrim and oblivion records.

And here I am again explaining for the millionth time that I can't work with code that diverges from upstream.

Another 5 minutes lost - are you happy ?

**Comment by Sharlikran** [comment]
> You even created a new dev with commits that are not in this dev

I did that to make sure all of the changes that I was using for the standalone version was provided.  Nothing more.  I most certainly wasn't trying to be defiant.

> And despite what you said you went ahead and made all your edits in the copy pasted code from the Skyrim constants and records ...  I spend all my Friday comparing your branch with mine, deleting duplicate code with Skyrim and trying to see what changed - most of your stuff is already in dev like the edits in Skyrim and oblivion records.

When I diverge from your code it allows me to see things that are happening and things that need to be updated.  There is nothing wrong with that.  I then use your code so I am using your code base.  Then I provide the results.

As for rebasing by branches I originally made an honest attempt to rebase off of dev exactly as I had promised I would.  You even saw that I tried very hard to do exactly as you asked.  In fact until I saw that it didn't work I was very proud of my attempt and you used some of the code you felt comfortable with.  What I did after that is just what I needed to do to have a working version for my self for the time being.

> Another 5 minutes lost - are you happy ?

No. Because I don't mean to cause you any discontent and waste your time.  It just happens because there is too much excess testosterone and it's always a big pissing contest all the time.  This was a simple issue.  I just pointed out that there was a FormID that was incorrect.  Then I even read Python tutorials so I could provide the code.  It is very rough and I bet it could be done the same with fewer lines of code.

A simple report, regardless of whether or not everyone can be reproduce it, and in the end it accounts for a FormID that should be added to the Fid list in Wrye anyway.

Arthmoor is the master of modding, no doubts there.  Utumno, you are the master of programming Wrye and I don't have to repeat all that stuff I always say.  I am part of the xEdit team and while Zilav and Hlp may know more then me I am still quite knowledgeable.  I know the inner workings of records very well.  I am the only person that can fix corrupted plugins to a working state without using programs like TESVSnip.  Plugins that are repaired so well that if you looked at them in a Hex editor you could very it. CoT and SoT to name a few big ones.  For Skyrim SE I repaired a few SPGD records to help a few famous modders.  They know how to resolve it themselves now and we are all trying to report it to Bethesda as we speak.  I know how to port the records from Delphi to Python.  I understand enough code to help, sometimes.  

When I provide code unless it's probably needed but maybe not written as well as it could be.  Unless it's a record definition which those are pretty static and just need testing and refinement.  

With the three of us working together we can accomplish great things.  The drama and the repeating will lessen if not stop when you realize that for a teIam to work you have to remove the "I" from team.  That's not how it's spelled in the first place.

**Comment by Infernio** [comment]
The actual bug that was reported here was most likely fixed in (probably the Use Global flag) f0147827e0480797f3f59f723bd464fa237b296d.
